ts/indicators/atr.src.ts

```diff
@@ -0,0 +1,215 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from '../parts/Globals.js';
+
+declare global {
+    namespace Highcharts {
+        class AtrIndicator extends SmaIndicator {
+            public data: Array<AtrIndicatorPoint>
+            public yData: Array<Array<number>>;
+            public getValues(
+                series: Series,
+                params: AtrInidicatorParamsOptions
+            ): (boolean|IndicatorValuesObject);
+        }
+
+        interface AtrIndicatorOptions extends SmaIndicatorOptions {
+            params?: AtrInidicatorParamsOptions;
+        }
+
+        interface AtrInidicatorParamsOptions {
+            period?: number;
+        }
+
+        class AtrIndicatorPoint extends SmaIndicatorPoint {
+            public series: AtrIndicator
+        }
+
+        interface SeriesTypesDictionary {
+            atr: typeof AtrIndicator;
+        }
+    }
+}
+
+
+import U from '../parts/Utilities.js';
+var isArray = U.isArray;
+
+var seriesType = H.seriesType,
+    UNDEFINED: undefined;
+
+/* eslint-disable valid-jsdoc */
+// Utils:
+/**
+ * @private
+ */
+function accumulateAverage(
+    points: Array<Array<(number|Array<number>)>>,
+    xVal: Array<number>,
+    yVal: Array<Array<number>>,
+    i: number
+): void {
+    var xValue = xVal[i],
+        yValue = yVal[i];
+
+    points.push([xValue, yValue]);
+}
+
+/**
+ * @private
+ */
+function getTR(currentPoint: Array<number>, prevPoint: Array<number>): number {
+    var pointY = currentPoint,
+        prevY = prevPoint,
+        HL = pointY[1] - pointY[2],
+        HCp = prevY === UNDEFINED ? 0 : Math.abs(pointY[1] - prevY[3]),
+        LCp = prevY === UNDEFINED ? 0 : Math.abs(pointY[2] - prevY[3]),
+        TR = Math.max(HL, HCp, LCp);
+
+    return TR;
+}
+
+/**
+ * @private
+ */
+function populateAverage(
+    points: any,
```

Comment:
This should be `Array<[number|Array<number>]>` too.

Reply:
Yes, it isn't used in the function so I tried the simple fix :)